src/main/java/org/apache/sling/resourceresolver/impl/providers/stateful/AuthenticatedResourceProvider.java

```diff
@@ -52,7 +52,7 @@ public class AuthenticatedResourceProvider {
 
     private static final Logger logger = LoggerFactory.getLogger(ResourceResolverImpl.class);
 
-    public static final AuthenticatedResourceProvider UNAUTHENTICATED_PROVIDER = new AuthenticatedResourceProvider(null, false, null, null);
```

Comment:
My IDE complained and failed the tests because `null` is being passed into `@NotNull` constructor arguments. That's why I fixed this.